relayer/relays/parachain/beefy-listener.go

```diff
@@ -154,18 +158,18 @@ func (li *BeefyListener) doScan(ctx context.Context, beefyBlockNumber uint64) er
 	if err != nil {
 		return err
 	}
-
 	for _, task := range tasks {
-		// do final proof generation right before sending. The proof needs to be fresh.
-		task.ProofOutput, err = li.generateProof(ctx, task.ProofInput, task.Header)
-		if err != nil {
-			return err
+		paraNonce := (*task.MessageProofs)[0].Message.Nonce
+		modNonce := paraNonce % li.scheduleConfig.Num
+		var waitingPeriod uint64
+		if modNonce > li.scheduleConfig.ID {
+			waitingPeriod = modNonce - li.scheduleConfig.ID
+		} else {
+			waitingPeriod = li.scheduleConfig.ID - modNonce
 		}
```

Comment:
Please double check but the math below will produce the waiting period correctly for all input.
```suggestion
		paraNonce := (*task.MessageProofs)[0].Message.Nonce
	        waitingPeriod := (paraNonce + li.scheduleConfig.Num - li.scheduleConfig.ID) % li.scheduleConfig.Num
```

Reply:
Nice catch! Fixed in https://github.com/Snowfork/snowbridge/pull/1266/commits/e787143109cf5d4b71b9afb84c1acb20892b6a35